### PR TITLE
backend support for email with name

### PR DESCRIPTION
**Pull request by martinraison**
@raymondng @2is10

### CODE REVIEW

**Comment by raymondng** [comment]
lgtm

**Comment by martinraison** [comment]
@2is10 I couldn't find an official specification for this kind of email address - is that a Gmail-only thing?

**Comment by 2is10** [comment]
@martinraison, here’s the spec:
http://tools.ietf.org/html/rfc2822#section-3.4

File: kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala [hunk 1]
```diff
@@ -167,11 +167,17 @@ class EContactRepoImpl @Inject() (
   def getOrCreate(userId:Id[User], email: String, name: Option[String], firstName: Option[String], lastName: Option[String])(implicit session: RWSession):Try[EContact] = Try {
     if (userId == null || email == null) throw new IllegalArgumentException("userId and email cannot be null")
 
-    val parsedResult = EmailParser.parseAll(EmailParser.email, email)
-    if (!parsedResult.successful) throw new IllegalArgumentException(s"Invalid email: $email")
+    // Parsing "email-like" expressions containing a name, such as "Douglas Adams <doug@kifi.com>"
+    val emailWithNameRe = """\s*([^\s<][^<]*[^\s<])\s+<(.*)>""".r
+    val (address, emailName) = emailWithNameRe.findFirstMatchIn(email) map { m =>
```
Comment (2is10) [inline kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala:172]: It’s a little confusing to be extracting a name from the `email` parameter to this method, given that there are already `name`, `firstName`, and `lastName` parameters too.

I don’t think we should be trying to parse names everywhere we expect an email address — only when we’re specifically parsing a recipient string or list (e.g. To: or Cc: field). Is there somewhere else we could be doing this parsing that’s closer to the point in the code where the extension posts a message with a recipient list?

**Pull request merged**